docs/api/navigation-history.md

```diff
@@ -74,3 +74,20 @@ Returns `boolean` - Whether the navigation entry was removed from the webContent
 #### `navigationHistory.getAllEntries()`
 
 Returns [`NavigationEntry[]`](structures/navigation-entry.md) - WebContents complete history.
+
+#### `navigationHistory.restore(index, entries)`
+
+Restores navigation history. Will make a best effort to restore not just the navigation stack but
+also the state of the individual pages - for instance including HTML form values or the scroll
+position. It's recommended to call this API before any navigation entries are created, so ideally
+before you call `loadURL()` or `loadFile()` on the `webContents` object.
+
+This API allows you to create common flows that aim to restore, recreate, or clone other webContents.
+
+* `offset` Integer
+* `entries` [NavigationEntry[]](structures/navigation-entry.md)
+
+Returns `Promise<void>` - the promise will resolve when the page has finished loading the selected navigation entry
+(see [`did-finish-load`](web-contents.md#event-did-finish-load)), and rejects
+if the page fails to load (see
+[`did-fail-load`](web-contents.md#event-did-fail-load)). A noop rejection handler is already attached, which avoids unhandled rejection errors.
```

Comment:
would be great to have a fiddle example for closing and restoring the window

Reply:
As a follow up, maybe a mermaid diagram to demonstrate where and when to insert history.

Reply:
We could have this in here: https://electronjs.org/docs/tutorial/navigation-history